src/app/_components/PillButton/component.js

```diff
@@ -1,21 +1,21 @@
+import React from 'react';
 import PropTypes from 'prop-types';
-import { cn } from '@/app/utils/cn';
-import { Paragraph } from '../Typography';
+import { Paragraph } from '@components';
+import { cn } from '@utils/cn';
+import { buttonColorVariant, buttonType } from './style';
 
-export function PillButton({
-  children,
-  type = 'button',
-  className,
-  variant,
-  colorVariant,
-  icon,
-  disabled = false,
-  onClick,
-}) {
-  const buttonVariant = icon ? variant?.icon : variant?.regular;
-  const { regular, hover, focused, active, disabled: disabledState } = colorVariant || {};
+export function PillButton({ children, className, icon, variant, colorVariant, ...props }) {
+  const buttonVariant = icon ? buttonType[variant]?.icon : buttonType[variant]?.regular;
+  const buttonColor = buttonColorVariant?.[variant]?.[colorVariant];
 
-  const { buttonStyle, layoutStyle } = buttonVariant || {};
+  if (!buttonVariant) {
+    throw new Error('Invalid button variant');
+  } else if (!buttonColor) {
+    throw new Error('Invalid button color variant');
+  }
```

Comment:
Dont throw an error, in case variant not found just ignore it and don't set buttonStyle and layoutStyle. For variants you could use https://cva.style/docs lib instead

Reply:
Thanks for comment! Fixed error throwing, now styles are not set if type or colorVariant isn't defined. About `cva` library, as i see its almost the same functionality as one we have now, not really sure why we need it.